src/modules/adaptors/Morpho/MorphoAaveV2ATokenAdaptor.sol

```diff
@@ -0,0 +1,175 @@
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Aave V2 aToken Adaptor V 1.1"));
+    }
+
+    /**
+     * @notice The Morpho Aave V2 contract on Ethereum Mainnet.
+     */
+    function morpho() internal pure returns (IMorpho) {
+        return IMorpho(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0);
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice Cellar must approve Morpho to spend its assets, then call supply to lend its assets.
+     * @param assets the amount of assets to lend on Morpho
+     * @param adaptorData adaptor data containing the abi encoded aToken
+     * @dev configurationData is not used for deposits bc it only influences user withdraws
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        // Deposit assets to Morpho.
+        IAaveToken aToken = abi.decode(adaptorData, (IAaveToken));
+        ERC20 underlying = ERC20(aToken.UNDERLYING_ASSET_ADDRESS());
+        underlying.safeApprove(address(morpho()), assets);
+
+        morpho().supply(address(aToken), assets);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(underlying, address(morpho()));
+    }
+
+    /**
+     * @notice Cellars must withdraw from Morpho
+     * @dev Important to verify that external receivers are allowed if receiver is not Cellar address.
+     * @param assets the amount of assets to withdraw from Morpho
+     * @param receiver the address to send withdrawn assets to
+     * @param adaptorData adaptor data containing the abi encoded aToken
+     * @param configData abi encoded bool indicating whether the position is liquid or not
+     */
+    function withdraw(
+        uint256 assets,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configData
+    ) public override {
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        // Make sure position is setup to be liquid.
+        bool isLiquid = abi.decode(configData, (bool));
+        if (!isLiquid) revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        IAaveToken aToken = abi.decode(adaptorData, (IAaveToken));
+
+        // Withdraw assets from Morpho.
+        morpho().withdraw(address(aToken), assets, receiver);
+    }
+
+    /**
+     * @notice Uses configuration data to determine if the position is liquid or not.
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory configData
+    ) public view override returns (uint256) {
+        bool isLiquid = abi.decode(configData, (bool));
+        if (!isLiquid) return 0;
```

Comment:
There exist a function in Morpho-AaveV2 Lens that can calculate the withdrawable amount: https://github.com/morpho-org/morpho-v1/blob/0df9d1a925d2342ea87f2fc5620d693604d29f3b/src/aave-v2/lens/UsersLens.sol#L53

Reply:
Oh this is great! I'll take a look and see if I can convert the math so that I can input a minimum health factor, and get an amount of assets withdrawable that will bring you down to that minimum.